integrations/llama_index/src/ucai_llamaindex/toolkit.py

```diff
@@ -169,3 +178,46 @@ def tools(self) -> List[FunctionTool]:
             List[FunctionTool]: A list of tools available in the toolkit.
         """
         return list(self.tools_dict.values())
+
+
+def extract_properties(data: dict) -> dict:
+    """
+    Extracts the 'properties' dictionary from the input dictionary,
+    merges its key-value pairs into the top-level dictionary, and returns a new dictionary.
+
+    Args:
+        data (dict): The original dictionary possibly containing a 'properties' key.
+
+    Returns:
+        dict: A new dictionary with 'properties' merged into the top-level.
+
+    Raises:
+        TypeError: If 'properties' exists but is not a dictionary.
+        KeyError: If there are key collisions between 'properties' and the top-level keys.
+    """
+    if not isinstance(data, dict):
+        raise TypeError("Input must be a dictionary.")
```

Comment:
```suggestion
        raise TypeError("Input must be a dictionary. {type(data).__name__}")
```

The actual data type should be included. It helps debugging.